File: dart/prime_sieve.dart
int primeSieve(int n) {
  List<bool> isPrime = List.filled(n + 1, true);
  isPrime[0] = isPrime[1] = false;
  
  for (int i = 2; i * i <= n; i++) {
    if (isPrime[i]) {
      for (int j = i * i; j <= n; j += i) {
        isPrime[j] = false;
      }
    }
  }
  
  int count = 0;
  for (int i = 2; i <= n; i++) {
    if (isPrime[i]) count++;
  }
  
  return count;
}

void main(List<String> args) {
  if (args.isEmpty) {
    print('Usage: prime_sieve <n>');
    return;
  }
  
  final n = int.parse(args[0]);
  
  final stopwatch = Stopwatch()..start();
  final count = primeSieve(n);
  stopwatch.stop();
  
  print('Dart: prime_sieve($n) = $count');
  print('Time: ${stopwatch.elapsedMilliseconds}ms');
}

File: rust/prime_sieve.rs
use std::env;
use std::time::Instant;

fn prime_sieve(n: usize) -> usize {
    let mut is_prime = vec![true; n + 1];
    is_prime[0] = false;
    if n > 0 {
        is_prime[1] = false;
    }
    
    let mut i = 2;
    while i * i <= n {
        if is_prime[i] {
            let mut j = i * i;
            while j <= n {
                is_prime[j] = false;
                j += i;
            }
        }
        i += 1;
    }
    
    let mut count = 0;
    for i in 2..=n {
        if is_prime[i] {
            count += 1;
        }
    }
    
    count
}

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() < 2 {
        println!("Usage: prime_sieve <n>");
        return;
    }
    
    let n: usize = args[1].parse().unwrap();
    
    let start = Instant::now();
    let count = prime_sieve(n);
    let duration = start.elapsed();
    
    let time_ms = duration.as_millis();
    
    println!("Rust: prime_sieve({}) = {}", n, count);
    println!("Time: {}ms", time_ms);
}